src/pykeen/pipeline.py

```diff
@@ -250,16 +253,153 @@ def title(self) -> Optional[str]:  # noqa:D401
+        if ax is None:
+            import matplotlib.pyplot as plt
+            ax = plt.gca()
+
         import seaborn as sns
-        sns.set()
-        plt.ylabel('Loss')
-        plt.xlabel('Epoch')
-        if self.title is not None:
-            plt.title(self.title)
-        return sns.lineplot(x=range(len(self.losses)), y=self.losses)
+        sns.set_style('darkgrid')
+
+        rv = sns.lineplot(x=range(len(self.losses)), y=self.losses, ax=ax)
+
+        loss_name = normalize_string(self.model.loss.__class__.__name__, suffix=_LOSS_SUFFIX)
+        ax.set_ylabel(f'{loss_name} Loss')
+        ax.set_xlabel('Epoch')
+        ax.set_title(self.title if self.title is not None else 'Losses Plot')
+        return rv
+
+    def plot_er(  # noqa: C901
+        self,
+        model: Optional[str] = None,
+        margin: float = 0.4,
+        ax=None,
+        entities: Optional[Set[str]] = None,
+        relations: Optional[Set[str]] = None,
+        apply_limits: bool = True,
+        plot_entities: bool = True,
+        plot_relations: bool = False,
+        annotation_x_offset: float = 0.02,
+        annotation_y_offset: float = 0.03,
+        **kwargs,
+    ):
+        """Plot the reduced entities and relation vectors in 2D.
+
+        :param model: The dimensionality reduction model from :mod:`sklearn`. Defaults to PCA.
+            Can also use KPCA, GRP, SRP, TSNE, LLE, ISOMAP, MDS, or SE.
+        :param kwargs: The keyword arguments passed to `__init__()` of
+            the reducer class (e.g., PCA, TSNE)
+
+        .. warning::
+
+            Plotting relations and entities on the same plot is only
+            meaningful for translational distance models like TransE.
+        """
+        if not plot_entities and not plot_relations:
+            raise ValueError
+
+        if model is None:
+            model = 'PCA'
+
+        reducer, reducer_kwargs = _get_reducer_cls(model, **kwargs)
+
+        if plot_relations and reducer.__name__ not in REDUCER_RELATION_WHITELIST:
+            raise ValueError(f'Can not use reducer {reducer} when projecting relations. Will result in nonsense')
+
+        if ax is None:
+            import matplotlib.pyplot as plt
+            ax = plt.gca()
+
+        import seaborn as sns
+        sns.set_style('whitegrid')
+
+        # reduce entity embedding dimensionality
+        e_emb = self.model.entity_embeddings.weight.detach().numpy()
+        if e_emb.shape[1] != 2:
+            entity_reduction_model = reducer(n_components=2, **reducer_kwargs)
+            e_emb_red = entity_reduction_model.fit_transform(e_emb)
+        else:
+            logger.debug('not reducing entity embeddings, already dim=2')
+            e_emb_red = e_emb
+
+        # reduce relation embedding dimensionality
+        r_emb = self.model.relation_embeddings.weight.detach().numpy()
+        if r_emb.shape[1] != 2:
+            relation_reduction_model = reducer(n_components=2, **reducer_kwargs)
```

Comment:
Here, we create a separate reducer object to project the relations.

Reply:
so you think we should use the pre-trained reducer from the entities?

Reply:
Yes.